src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java

```diff
@@ -713,6 +715,20 @@ public void flush() throws CredentialStoreException {
         }
     }
 
+    /**
+     * Returns credential aliases stored in this store as {@code Set<String>}.
+     * <p>
+     * It is not mandatory to override this method (throws {@link UnsupportedOperationException} by default).
+     *
+     * @return {@code Set<String>} of all keys stored in this store
+     * @throws UnsupportedOperationException when this method is not supported by the underlying credential store
+     * @throws CredentialStoreException      if there is any problem with internal store
+     */
+    @Override
+    public Set<String> getAliases() throws UnsupportedOperationException, CredentialStoreException {
+        return cache.keySet();
```

Comment:
This probably should have either made a copy or made it unmodifiable.  I've made a note to fix this later.